### ISSUE TITLE
Output of anova() prints twice at REPL

### ISSUE THREAD

**Issue by BioTurboNick** [marcpabst/ANOVA.jl#6]
The output of anova() at the REPL prints its output normally first, then repeats it after the `Julia>` prompt interrupting the first line. Typing at the prompt moves the prompt to the last line, overwriting it.
<img width="505" alt="anova" src="https://user-images.githubusercontent.com/1438610/49681680-03b2cd00-fa74-11e8-8c82-ccb3ce8c051b.PNG">
This is in JuliaPro 1.0.1.1 with Julia 1.0.2

**Comment by marcpabst** [comment]
That's weird. I'm just using a simple DataFrame to print the AnovaObject:

```
function Base.show(io::IO, x::AnovaObject)
  print(DataFrame( 
          Source = x.Source,
          DF = x.DF,
          SS = x.SS,
          MSS = x.MSS,
          F = x.F,
          p = x.p
          ))
end
```

The only thing I can think of is that DataFrame itself creates the output. I currently don't have a Julia output installed but will look into it when I have more time. Thanks for the issue!

**Comment by BioTurboNick** [comment]
It's possible it's a more general issue with the REPL or Juno. For instance, I've noticed that when I return a plot from a function, it may render twice. I might poke around more.

**Comment by marcpabst** [comment]
Have you been able to solve your issue?

**Comment by BioTurboNick** [comment]
Yes, it appears that the issue is specific to the Juno REPL. It does not occur when Julia REPL is run in CMD.exe.

I'm closing this and I'll report it for Juno.

**Comment by pfitzseb** [comment]
Hm, the problem isn't really with Juno here but with your `show` method  -- it's expected that you actually print something to `io` instead of `stdout`, i.e.
```
function Base.show(io::IO, x::AnovaObject)
  print(io, DataFrame( 
          Source = x.Source,
          DF = x.DF,
          SS = x.SS,
          MSS = x.MSS,
          F = x.F,
          p = x.p
          ))
end
```

I'm pretty confident that making this change will fix the weird behaviour in Juno/JuliaPro.

**Comment by BioTurboNick** [comment]
Ah. Why would the behavior differ between the two REPLs?

**Comment by pfitzseb** [comment]
Not sure, tbh. I can't repro this when evaling in the REPL only.
Can you try 
```
struct Foo end

function Base.show(io::IO, x::Foo)
  print("hii")
  sleep(0.5)
  print("ho")
end

Foo()
```
in the REPL and in an editor for me?

**Comment by BioTurboNick** [comment]
In Juno:
```
julia> Foo()
hiihiiho

julia>ho
```

Where "ho" is just displayed at the prompt but not really there.

**Comment by pfitzseb** [comment]
The `show` method should still be fixed in this package though, @marcpabst.

**Comment by marcpabst** [comment]
Thanks, @pfitzseb for exposing the problem! I'll make the change within the next few days!

**Comment by Nosferican** [comment]
Will be fixed as part of #10.
